common/txmgr/types/tx.go

```diff
@@ -109,3 +112,192 @@ type TxMeta[ADDR types.Hashable, TX_HASH types.Hashable] struct {
 }
+
+type TxAttempt[
+	CHAIN_ID ID,
+	ADDR types.Hashable,
+	TX_HASH, BLOCK_HASH types.Hashable,
+	R ChainReceipt[TX_HASH],
+	FEE Fee,
+	ADD any,
+] struct {
+	ID    int64
+	TxID  int64
+	Tx    Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]
+	TxFee FEE
+	// ChainSpecificFeeLimit on the TxAttempt is always the same as the on-chain encoded value for fee limit
+	ChainSpecificFeeLimit   uint32
+	SignedRawTx             []byte
+	Hash                    TX_HASH
+	CreatedAt               time.Time
+	BroadcastBeforeBlockNum *int64
+	State                   TxAttemptState
+	Receipts                []Receipt[R, TX_HASH, BLOCK_HASH] `json:"-"`
+	TxType                  int
+}
+
+func (a TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) String() string {
+	return fmt.Sprintf("TxAttempt(ID:%d,TxID:%d,Fee:%s,TxType:%d", a.ID, a.TxID, a.TxFee, a.TxType)
+}
+
+func (a TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) Fee() FEE {
+	return a.TxFee
+}
+
+func (a TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) GetBroadcastBeforeBlockNum() *int64 {
+	return a.BroadcastBeforeBlockNum
+}
+
+func (a TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) GetChainSpecificFeeLimit() uint32 {
+	return a.ChainSpecificFeeLimit
+}
+
+func (a TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) GetHash() TX_HASH {
+	return a.Hash
+}
+
+func (a TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD]) GetTxType() int {
+	return a.TxType
+}
+
+type Tx[
+	CHAIN_ID ID,
+	ADDR types.Hashable,
+	TX_HASH, BLOCK_HASH types.Hashable,
+	R ChainReceipt[TX_HASH],
+	FEE Fee,
+	ADD any,
+] struct {
+	ID             int64
+	Sequence       *int64
+	FromAddress    ADDR
+	ToAddress      ADDR
+	EncodedPayload []byte
+	Value          big.Int
+	// FeeLimit on the Tx is always the conceptual gas limit, which is not
+	// necessarily the same as the on-chain encoded value (i.e. Optimism)
+	FeeLimit uint32
+	Error    null.String
+	// BroadcastAt is updated every time an attempt for this eth_tx is re-sent
+	// In almost all cases it will be within a second or so of the actual send time.
+	BroadcastAt *time.Time
+	// InitialBroadcastAt is recorded once, the first ever time this eth_tx is sent
+	InitialBroadcastAt *time.Time
+	CreatedAt          time.Time
+	State              TxState
+	TxAttempts         []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD] `json:"-"`
+	// Marshalled TxMeta
+	// Used for additional context around transactions which you want to log
+	// at send time.
+	Meta    *datatypes.JSON
+	Subject uuid.NullUUID
+	ChainID CHAIN_ID
+
+	PipelineTaskRunID uuid.NullUUID
+	MinConfirmations  clnull.Uint32
+
+	// AdditionalParameters is generic type that supports passing miscellaneous parameters
+	// as a part of the TX struct that may be used inside chain-specific components
+	// example: EVM + NullableEIP2930AccessList which only affects on DynamicFee transactions
+	AdditionalParameters ADD
```

Comment:
it's currently used in the evm/txmgr/attempts.go file for dynamic fee transactions :)

Reply:
Actually, even there, it is only used if a caller sends us an AccessList value.
At present, there's no caller sending that.
So as per today's use-cases, we could completely delete that easily.

But perhaps we won't do, in case we need to use it in future.